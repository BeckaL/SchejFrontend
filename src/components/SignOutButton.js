import React, { Component } from 'react'
import axios from 'axios'
import { deleteAuthStorage } from '../actions/authentication'

export default class SignOutButton extends Component {

  handleClick = () => {
    let self = this
    axios.delete(
      `${process.env.REACT_APP_API_URL}/sign_out`,
      {
        authentication_token: localStorage['authenticationToken']
      }
    )
    .then(response => {
      deleteAuthStorage()
      self.props.history.push('/')
    })
    .catch(error => {
      console.log(error)
    })
  }

  render() {
    return (
      <button
        id='sign-out-button'
        className='custom-button'
        onClick={this.handleClick}
      >
        Sign Out
      </button>
    )
  }
}
