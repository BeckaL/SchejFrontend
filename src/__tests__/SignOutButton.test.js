import React from 'react'
import axios from 'axios'
import { shallow } from 'enzyme'
import SignOutButton from '../components/SignOutButton'

jest.mock('axios')

describe('Sign out button', () => {
  var wrapper

  beforeAll(() => {
    wrapper = shallow(<SignOutButton history={[]}/>)
  })

  test('it matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('it renders a button', () => {
    const buttons = wrapper.find('button')
    expect(buttons.length).toEqual(1)
  })

  test('it makes the correct API call on click', () => {
    axios.delete.mockResolvedValue({})

    const button = wrapper.find('button')
    button.simulate('click')

    expect(axios.delete).toHaveBeenCalledTimes(1)
    expect(axios.delete).toHaveBeenCalledWith(
      'http://localhost:3001/api/v1/sign_out', {
        authentication_token: undefined
      }
    )
  })
})
