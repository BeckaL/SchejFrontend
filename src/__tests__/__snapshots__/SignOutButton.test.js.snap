// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign out button it matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignOutButton
    history={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Sign Out",
      "className": "custom-button",
      "id": "sign-out-button",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Sign Out",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Sign Out",
        "className": "custom-button",
        "id": "sign-out-button",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Sign Out",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
